### DISCUSSION TITLE
SX1276 state of DIOx lines in Sleep Mode? Tri-state / Hi-Z?

### DISCUSSION THREAD

**Discussion by pieterconradie** [Lora-net/LoRaMac-node#1007] [Q&A]
Hi there,

In the SX1276 datasheet, the DIOx pins are listed as "Digital I/O, software configured", but in figure 25 on p. 65, DIO0, DIO1, DIO3, DIO4 and DIO5 are indicated as outputs only and DIO2 as input/output. I can't find definitive documentation on these pins in the datasheet.

1. Are they PUSH-PULL outputs all the time? E.g. the pins are never inputs / tri-state / hi-Z?
2. Do they stay PUSH-PULL outputs even when the SX1276 is placed into Sleep Mode?

This is important to know, because any floating inputs on the microcontroller, e.g. STM32L072 inside Murata Type ABZ module will cause increased power consumption. Also, I do not want to enable an internal pull-up because that will also lead to increased power consumption if it is pulled down.

I think / suspect that they are PUSH-PULL outputs all the time but why:

a. does the SX1276IoInit() function configure the STM32 pins connected to DIOx as inputs with pull-ups enabled?
b. does the SX1276IoDeInit() function configure the STM32 pins as inputs with pull-ups disabled?

See here:
https://github.com/Lora-net/LoRaMac-node/blob/master/src/boards/B-L072Z-LRWAN1/sx1276-board.c

Thanks in advance,
Pieter
https://piconomix.com

**Comment by mluis1** [comment]
I need some more time to answer you. It has been a few years now that I didn't look on how the DIO pins are setup in the different modes.
Back then we have verified that all provided examples were using less than 2 uA when the MCU and radio were set in low-power modes. The MCU pins configuration provided in this project is supposed to achieve that goal.  

In the future it would be nice if you could post this kind of questions on the project [Discussions](https://github.com/Lora-net/LoRaMac-node/discussions) tab. It is a better place to engage discussions and then we can agree if it is an issue or not.
If you don't mind I move this issue to the [Discussions](https://github.com/Lora-net/LoRaMac-node/discussions) tab.

**Reply by mluis1** [reply]
Unfortunately, I have not find more information on the subject.

I would recommend to post your question on below forum in order to get better support on Semtech radios usage.

https://forum.lora-developers.semtech.com/

**Reply by pieterconradie** [reply]
Hi @mluis1 

Thanks so much for taking the time and care to reply (and moving my post to the right area).

Kind regards,
Pieter
